dev-python/rdflib/rdflib-4.2.2.recipe

```diff
@@ -0,0 +1,95 @@
+SUMMARY="Python library for working with RDF"
+DESCRIPTION="RDFLib is a Python library for working with RDF, a simple \
+yet powerful language for representing information as graphs."
+HOMEPAGE="https://github.com/RDFLib/rdflib"
+COPYRIGHT="2002-2017 RDFLib Team"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="http://github.com/RDFLib/rdflib/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="a362926f77624a8abfb93042c28607f49445d003792d5271cd07f0610ab16033"
+
+ARCHITECTURES="any"
+
+PROVIDES="
+	rdflib = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PYTHON_PACKAGES=(python)
+PYTHON_VERSIONS=(2.7 3.6)
+for i in "${!PYTHON_PACKAGES[@]}"; do
+	pythonPackage=${PYTHON_PACKAGES[i]}
+	pythonVersion=${PYTHON_VERSIONS[$i]}
+
+	eval "PROVIDES_$pythonPackage=\"
+		${portName}_$pythonPackage = $portVersion
+		cmd:csv2rdf
+		cmd:rdf2dot
+		cmd:rdfgraphisomorphism
+		cmd:rdfpipe
+		cmd:rdfs2dot
```

Comment:
@korli @fbrosson This package provides `cmd:`s for both `_python` and `_python3` package. How should we resolve the conflict when an user want to install both packages?

Reply:
@fbrosson I might need help...
The `if` statement on line 87 (the INSTALL function) is causing problems for me... I tried following the examples (tried both renaming and removing), but HaikuPorter keeps telling me that I can't remove or rename the directory `$binDir/rdflib` because it doesn't exist.

Reply:
@bach5000: You need to find out what gets really installed in each subpackage (without any deletions or renames) before you can decide what you need to remove or rename.

Reply:
@fbrosson: oops. just figured out what I did wrong for my first problem :)
But what about the policy errors? I put the bins in PROVIDES for python... why can't it find them?

Reply:
@fbrosson Should I remove the binaries (like coverage) or should I rename them (like pbr)? What do you think?

Reply:
It depends:
* on which version of python you want the "non-suffixed" entry point to use,
* and on what gets installed when there are no renames nor deletes.

Imagine you would like the "non-suffixed" entry point to use python3. If the python (i.e. python2) subpackage comes with "foobar" and "foobar2" (and if they are equivalent entry points) then you ~can~ _should_ delete "foobar" in the python subpackage.

If, for some reason, you prefer the "non-suffixed" entry point to use python 2 then you will need to:
* rename "foobar" to "foobar3" in the python3 subpackage (if it does not already install a "foobar3").
* delete "foobar" in the python3 subpackage (if it already comes with a "foobar3").

Last "rule": if the python module cannot be built with gcc2 then on x86_gcc2 there won't be any python (i.e. python 2) subpackage, so the "non-suffixed" entry point should probably use python3.
BTW, in that case you need to enable x86 secondary arch and "populate" `PYTHON_PACKAGES`  and `PYTHON_VERSIONS` like in dev-python/coverage. That's the only way to have a 100% correct recipe in all cases because on x86_gcc2 non hybrid the recipe would not even be buildable.